### PR TITLE
Deprecate P2QR. Postpone SQIsign. Additional detail on Grover's.

### PR DESCRIPTION
**Pull request by cryptoquick**
This should be one of the final changes I'd like to make before submitting upstream to the bips repo.

### CODE REVIEW

File: bip-p2qrh.mediawiki
Comment (CrewsControlSolutions) [inline bip-p2qrh.mediawiki:17]: love it!

File: bip-p2qrh.mediawiki
Comment (CrewsControlSolutions) [inline bip-p2qrh.mediawiki:29]:
```suggestion
Mining may one day be vulnerable to disruption by very advanced quantum computers making use of Grover's algorithm. However, Grover's [https://arxiv.org/pdf/1902.02332 scales very poorly] compared to Shor's, requiring 10^40 quantum operations in comparison to 10^8 for running Shor's over ECC. It's for this reason that the primary threat to Bitcoin by quantum computers is to its signature algorithm and not Proof of Work, hence the focus on a new address format.
```

Also really like this, well said.  Sentence length is perfect and flows nicely.  I did remove the "It" in the 2nd sentence and replaced it with "Grover's" for clarification.

File: bip-p2qrh.mediawiki
Comment (CrewsControlSolutions) [inline bip-p2qrh.mediawiki:59]: Perfect

File: bip-p2qrh.mediawiki
Comment (CrewsControlSolutions) [inline bip-p2qrh.mediawiki:72]:
```suggestion
P2QRH is meant to be implemented on top of P2TR, combining the security of classical Schnorr signatures along with post-quantum cryptography. This is a form of "hybrid cryptography" such that no regression in security is presented should a vulnerability exist in one of the signature algorithms used. One key distinction between P2QRH and P2TR however is that P2QRH will encode a hash of the public key. This is a significant change in how Taproot works, but is necessary to avoid exposing public keys on-chain in advance of attackers.
```

Great sentence length and flow.

File: bip-p2qrh.mediawiki
Comment (CrewsControlSolutions) [inline bip-p2qrh.mediawiki:74]:
```suggestion
P2QRH uses a 32-byte HASH256 (specifically SHA-256 twice-over, which is similar to that used in [https://github.com/bitcoin/bips/blob/master/bip-0016.mediawiki#specification BIP-16]) of the public key to reduce the size of new outputs and also to increase security by not having the public key available on-chain. This hash serves as a minimal cryptographic commitment to a public key. It goes into the output spend script, which does not receive the witness discount.
```

Always good to be specific on what the "this" is referring to.

File: bip-p2qrh.mediawiki
Comment (CrewsControlSolutions) [inline bip-p2qrh.mediawiki:78]: Perfecto.

File: bip-p2qrh.mediawiki
Comment (CrewsControlSolutions) [inline bip-p2qrh.mediawiki:80]:
```suggestion
Support for FALCON signatures will be introduced first, with the intention of adding SQIsign and other post-quantum algorithms as they are approved. By way of comparison, FALCON signatures are roughly 4x larger than SQIsign signatures and 20x larger than Schnorr signatures. FALCON is a more conservative approach to applied lattice cryptography than SQIsign, and its use has recently been approved by NIST. However, even SQIsign signatures are roughly 5x larger than Schnorr signatures. This means, to maintain present transaction throughput, an increase in the witness discount will likely be desired in a QuBit soft fork. That will be specified in a future QuBit BIP.
```

File: bip-p2qrh.mediawiki
Comment (CrewsControlSolutions) [inline bip-p2qrh.mediawiki:82]:
```suggestion
An increase in the witness discount must not be taken lightly.  It must be resistant to applications that might take advantage of this discount (e.g. storage of arbitrary data as seen with "inscriptions") without a corresponding increase in economic activity. Such an increase would not only impact node runners but those with inscriptions would also have the scarcity of their non-monetary assets affected. The only way to prevent this while also increasing the discount is to have a completely separate witness, a quantum witness, or "quitness," that is solely responsible for providing post-quantum signatures.
```

Nice

File: bip-p2qrh.mediawiki
Comment (CrewsControlSolutions) [inline bip-p2qrh.mediawiki:86]:
```suggestion
Since only valid signatures can be committed to a SegWit v3 quitness, arbitrary data cannot be added by miners, as that would affect the consensus of their block. A CRQC operator is economically disincentivized from computing a spendable public key that matched arbitrary signature data due to the cost of that computation. That is because the cost of such a computation could prove quite substantial, rather than simply putting the arbitrary data within a Taproot witness. Doing the work to meet the requirement for it to be consensus-valid data would prove cost-prohibitive.
```

File: bip-p2qrh.mediawiki
Comment (CrewsControlSolutions) [inline bip-p2qrh.mediawiki:90]:
```suggestion
While it might be seen as a maintenance burden for bitcoin ecosystem devs to go from a single cryptosystem implementation to two distinct cryptosystems-- and it most certainly is-- the ramifications of a chain broken through extrinsic factors should provide sufficient motivation. An increase in software maintenance everywhere signatures are used should be seen as an acceptable compromise for maintained integrity of bitcoin transfers during a regime of quantum advantage.
```

File: bip-p2qrh.mediawiki
Comment (CrewsControlSolutions) [inline bip-p2qrh.mediawiki:155]: Excellent

File: bip-p2qrh.mediawiki
Comment (CrewsControlSolutions) [inline bip-p2qrh.mediawiki:166]:
```suggestion
Ideally FALCON and SQIsign prove to be flexible enough to support [https://www.pierrickdartois.fr/homepage/wp-content/uploads/2022/04/Report_OSIDH_DARTOIS.pdf Isogeny Diffie-Hellman] to replace ECDH applications and also provide methods for the key tweaking necessary to support TapScript for P2QR addresses. Additionally, isogeny-based post-quantum cryptography is based on higher-order elliptic curves, and so it might be possible to implement Isogeny Schnorr signatures.
```

File: bip-p2qrh.mediawiki
Comment (CrewsControlSolutions) [inline bip-p2qrh.mediawiki:217]: Do we need to make clear that we are deprecating P2QR in favor of having only P2QRH?  I mean, we do if you read the BIP, but at a high level glance you might not know.  And I guess that would be the reader's loss.

**Comment by CrewsControlSolutions** [comment]
Nice work.  Looking at the BIP from a high-level and not specific to the PR, should we have the 9 other sections (Specification, Validation, Test Vectors, etc.) completed by the time we formally submit the BIP?  Or is it better to hold off on those sections until we formally submit the BIP?  The latter may provide more technical feedback like you mentioned before.

**Review by CrewsControlSolutions — APPROVED** [review]
Approve w/ changes

**Comment by CrewsControlSolutions** [comment]
I still need to do a once-over of the BIP to make sure things look good before formal BIP submission.  It's too much to bundle that with this PR.  Once this PR is merged and after our conversation with Jeff, I will do my final review and create a PR with my suggestions.

File: bip-p2qrh.mediawiki
Comment (cryptoquick) [reply bip-p2qrh.mediawiki:86]: Actually, "in" is the appropriate phrasing here. For example, where are Taproot signatures stored? In the witness.

File: bip-p2qrh.mediawiki
Comment (cryptoquick) [reply bip-p2qrh.mediawiki:166]: FALCON does not support isogeny cryptography so far as I know.

Comment (CrewsControlSolutions) [reply bip-p2qrh.mediawiki:166]: Gotcha. I was wondering that. I think this paragraph is fine as is then.

But I feel like this section after the table of signature algorithms section leads the reader to think that SQIsign is the preferred option.  Do we need to have more discussion about why FALCON is the number one option?

**Comment by cryptoquick** [comment]
I think we should present this to the broader community before getting more specific and introducing actual test vectors.

**Pull request merged**
